src/mympi.cpp

```diff
@@ -259,10 +287,10 @@ int max_to_all(int in) {
 }
 
 ivec max_to_all(const ivec &pt) {
-  int in[5], out[5];
+  long int  in[5], out[5];
```

Comment:
As long as `sizeof(long int) >= sizeof(meep::integer)` everything is fine because the data of `pt` is copied. However, on systems with LLP64 data model (are there any except MS Windows?) there will be trouble (maybe also in other places?). 
One way to resolve the issue would be to use `long long int` instead of `long int`. 
Another way would be to use `meep::integer` and define a suiteable MPI data type, e.g. `MPI_MEEP_INTEGER`, that points either to `MPI_INT`, `MPI_LONG` or `MPI_LONG_LONG_INT`, depending on the size of `meep::integer`. Then it would be possible to write
```c++
ivec max_to_all(const ivec &pt) {
    meep::integer  in[5], out[5];
    for (int i=0; i<5; ++i) in[i] = out[i] = pt.in_direction(direction(i));
 #ifdef HAVE_MPI
    MPI_Allreduce(&in,&out,5,MPI_MEEP_INTEGER,MPI_MAX,mycomm);
 #endif
   ivec ptout(pt.dim);
   for (int i=0; i<5; ++i) ptout.set_direction(direction(i), out[i]);
   return ptout;
 }
```
I will try this and see how far I can get with `AC_CHECK_SIZEOF` and the preprocessor.